ocaml/xapi/xapi_session.ml

```diff
@@ -1195,6 +1196,12 @@ let local_logout ~__context =
   let session_id = Context.get_session_id __context in
   Xapi_local_session.destroy ~__context ~self:session_id
 
+let assert_username uname = 
+  match uname with
+  | "" ->
+    failwith ("Username is required to login")
```

Comment:
It would be better to raise an API error here. The error `session_authentication_failed` looks suitable. So:

    raise (Api_errors.(Server_error (session_authentication_failed, []))

Reply:
How about  `raise (Api_errors.(Server_error (session_authentication_failed, ["Username is required to login"])))`   which will deliver an explicit error message to the client. CC @liulinC

Reply:
See below: we need to consider the error definition in datamodel_errors.ml.